pkg/build/controller/build/defaults/defaults.go

```
@@ -218,18 +226,20 @@ func addDefaultAnnotation(k, v string, annotations map[string]string) {
 	}
 }
 
-func addDefaultToleration(t kapi.Toleration, tolerations *[]v1.Toleration) {
+func addDefaultToleration(t kapi.Toleration, tolerations *[]v1.Toleration) error {
 	tol := v1.Toleration{}
 
 	if err := kapiv1.Convert_core_Toleration_To_v1_Toleration(&t, &tol, nil); err != nil {
-		panic(err)
+		utilruntime.HandleError(fmt.Errorf("Unable to convert core.Toleration to v1.Toleration: %v", err))
```

you should return the same error you're logging.